packages/block-editor/src/components/rich-text/use-enter.js

```diff
@@ -67,7 +68,14 @@ export function useEnter( props ) {
 
 			const { text, start, end } = _value;
 
-			if ( event.shiftKey ) {
+			if ( shouldDisableEditing ) {
```

Comment:
Can you explain what that branch of the code is doing?

Reply:
I believe this explanation about how `splitValue` works should help to understand this as well: [link](https://github.com/WordPress/gutenberg/pull/58958/files#r1500543713).

Another option here could be to change `_value.end = _value.text?.length`.

From my testing, it seems that when the rich text is not editable, it acts as if the cursor was in the initial character. This causes that when you click on enter, it doesn't work as expected. For that reason, I believe we could assume that when `shouldDisableEditing` is true, we don't have to add any text to the new block. Or we want to simulate placing the cursor at the end.

Reply:
I see. I was totally lost, so that made it clear. I think simulating the case that the cursor is at the end should do the trick for most of the blocks 💯

Reply:
Just changed that part in [this commit](https://github.com/WordPress/gutenberg/pull/58958/commits/c25a78588da9e4e35c4ba53ea29374a129b79bac).